snake-odin/src/snake.odin

```diff
@@ -0,0 +1,109 @@
+package main
+
+import "w4"
+import "core:runtime"
+
+Point :: struct {
+	X: i32,
+	Y: i32,
+}
+
+are_points_colliding :: proc "c" (a, b : Point) -> bool {
+	return a.X == b.X && a.Y == b.Y 
+}
+
+Snake :: struct {
+	body: [20*20]Point,
+	direction: Point,
+	len: int,
+}
+
+reset_snake :: proc "c" (s : ^Snake) {
+	using s
+
+	len = 3
+	body[0] = Point{X = 2, Y = 0}
+	body[1] = Point{X = 1, Y = 0}
+	body[2] = Point{X = 0, Y = 0}
+	direction = Point{X = 1, Y = 0}
+}
+
+draw_snake :: proc "c" (s : ^Snake) {
+	using s
```

Comment:
I'd recommend against `using` and just write `s.` for all the things, especially since it is very short.

Reply:
Will do.